pootle/apps/pootle_store/models.py

```diff
@@ -514,21 +525,28 @@ def update(self, unit, user=None):
             User = get_user_model()
             user = User.objects.get_system_user()
 
-        if (self.source != unit.source or
-            len(self.source.strings) != stringcount(unit.source) or
-            self.hasplural() != unit.hasplural()):
-
+        update_source = (
+            self.source != unit.source
+            or (len(self.source.strings)
+                != stringcount(unit.source))
+            or (self.hasplural()
+                != unit.hasplural()))
+        if update_source:
             if unit.hasplural() and len(unit.source.strings) == 1:
                 self.source = [unit.source, PLURAL_PLACEHOLDER]
             else:
                 self.source = unit.source
-
             changed = True
 
-        if (self.target != unit.target or
-            len(self.target.strings) != stringcount(unit.target)):
+        update_target = (
+            self.target != unit.target
+            or (len(self.target.strings)
+                != stringcount(unit.target)))
```

Comment:
I know this is inherited code, but not sure I understand the need for the length checking.